client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java

```diff
@@ -203,19 +214,21 @@ public int getSpaceIDFromCache(String name) {
     }
 
     @Override
-    public Map<String, Map<Integer, List<HostAndPort>>> getParts() {
+    public Map<String, Map<Integer, List<HostAndPort>>> getPartsAllocFromCache() {
         return this.spacePartLocation;
     }
 
     @Override
-    public List<HostAndPort> getPart(String spaceName, int part) {
+    public List<HostAndPort> getPartAllocFromCache(String spaceName, int part) {
+        if (spacePartLocation.containsKey(spaceName)) {
+            Map<Integer, List<HostAndPort>> partsAlloc = spacePartLocation.get(spaceName);
+            if (partsAlloc.containsKey(part)) {
+                return partsAlloc.get(part);
+            }
+        }
         return null;
```

Comment:
return Lists.newLinkedList() would be better?

Reply:
I guess `null` implies that a failure, but an empty list would imply that no machine hold this part.